Library/Formula/osm2pgsql.rb

```diff
@@ -35,14 +28,18 @@ def install
       "--with-proj=#{Formula["proj"].opt_prefix}",
       "--with-boost=#{Formula["boost"].opt_prefix}",
       "--with-zlib=/usr",
-      "--with-bzip2=/usr",
-      # Related to the patch, remove this line when upgrading
-      "--without-lockfree"
+      "--with-bzip2=/usr"
     ]
     puts args
     if build.with? "protobuf-c"
       args << "--with-protobuf-c=#{Formula["protobuf-c"].opt_prefix}"
     end
+    # Mountain Lion has some problems with C++11.
+    # This is probably going to be a fatal issue for 0.89 and 0.90 on it, but
+    # for now it can be worked around
+    if MacOS.version < :mavericks
+      args << "--without-cxx11"
+    end
```

Comment:
> Another option is adding needs :cxx11 which will ensure it's using a CXX11 compiler.

This was generating linker errors beyond my ability to diagnose through Jenkins alone